wire/common.go

```diff
@@ -391,9 +403,21 @@ func ReadVarInt(r io.Reader) (uint64, error) {
 	return rv, nil
 }
 
-// WriteVarInt serializes val to w using a variable number of bytes depending
+// WriteVarIntLittleEndian serializes val to w using a variable number of bytes depending
+// on its value using little endian encoding.
+func WriteVarIntLittleEndian(w io.Writer, val uint64) error {
```

Comment:
I'm not sure I'm a big fan of having two functions, each for one endianness. Compare the following usage:
WriteVarIntLittleEndian(w, num) x 10
vs.
var byteOrder = LittleEndian
WriteVarInt(w, num, byteOrder) x 10

It's much faster to refactor the latter, and there's no sacrifice in readability.